### PR TITLE
feat(serenity): add Semrush-backed competitor-summary endpoint for Overview

### PR DESCRIPTION
**Pull request by aniham**
## Summary
- Adds `GET .../serenity/brand-presence/competitor-summary` — the lightweight, aggregate-totals counterpart to `market-tracking-trends`.
- Reuses the same two Semrush elements `market-tracking-trends.js` already queries (TRENDS_MV mentions + MARKET_CITATIONS_TREND citations), summed per-competitor instead of returned as a weekly series — no new Semrush query shape.
- Backs project-elmo-ui's Overview screen Competitor Comparison chart once it switches to Semrush (companion PR in adobe/project-elmo-ui).
- New files: `src/support/elements/definitions/competitor-summary.js` (transform), `getCompetitorSummary` in `elements-service.js` (service method) and `elements.js` (handler, modeled on `getMarketTrackingTrends`'s param resolution).
- Route classified in `required-capabilities.js` (`brand:read`) and `facs-capabilities.js` (`llmo/can_view`), matching every sibling `serenity/brand-presence/*` route.

## Test plan
- [x] `npm run lint` clean
- [x] `npm run type-check` clean
- [x] Full unit suite passes (15838 passing, 0 failing) after syncing `node_modules` to the pinned Node 24 / lockfile versions
- [x] `test/routes/facs-capabilities.test.js`, `test/routes/required-capabilities.test.js`, `test/routes/index.test.js`, `test/controllers/elements.test.js`, `test/support/elements/elements-service.test.js` all green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/support/elements/definitions/competitor-summary.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * project-elmo-ui's brandPresencePgApi.ts) rather than the heavier
+ * `market-tracking-trends` weekly-breakdown shape.
+ *
+ * @param {object} mentionsRaw - Raw TRENDS_MV response.
+ * @param {object} citationsRaw - Raw MARKET_CITATIONS_TREND response.
+ * @param {string} brandName - Tracked brand's display name (matches its `legend`;
+ *   excluded from the returned competitor list).
+ * @returns {{ competitors: Array<{ name: string, mentions: number, citations: number }> }}
+ */
+/* c8 ignore start -- competitor-summary POC endpoint; unit tests intentionally deferred */
```
Comment (MysticatBot) [inline src/support/elements/definitions/competitor-summary.js]: issue (blocking): The repo CLAUDE.md states: *Behavior changes must include unit tests.* `transformCompetitorSummary` is a pure function (Map input, object output) that is trivial to test in isolation. A ~20-line test covering: (1) happy path with both metrics summed correctly, brand excluded, sorted by mentions desc; (2) empty `blocks.lines` input would protect the non-obvious `y__mentions` field reuse (correct per the Semrush data model, but a future contributor could easily mistake it for a bug and read a different field for citations). At minimum, remove the `c8 ignore` on this transform and add a unit test in `test/support/elements/definitions/competitor-summary.test.js`. The controller and service-layer tests can remain deferred if the goal is to stabilize the upstream Semrush contract first.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @aniham,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - missing unit tests and OpenAPI spec for a new endpoint that will back a production UI.
**Complexity: HIGH** - medium diff; API surface + FACS classification change.
_Changes: adds a Semrush-backed `GET .../brand-presence/competitor-summary` endpoint returning aggregate per-competitor mentions/citations totals for the Overview chart (8 files)._

## Must fix before merge

1. [Important] Missing unit tests for `transformCompetitorSummary` - `src/support/elements/definitions/competitor-summary.js:34` (details inline)
2. [Important] Missing OpenAPI specification - CLAUDE.md mandates "OpenAPI First: Define API contract in `docs/openapi/` before implementation" (step 1 of Adding a New Endpoint)
3. [Important] Missing integration tests - CLAUDE.md mandates "New or modified endpoints must include integration tests in `test/it/`"

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- suggestion: Extract the shared auth/date-validation/region-resolution preamble (repeated verbatim in `getMarketTrackingTrends`, `getStats`, and now `getCompetitorSummary`) into a `resolveMarketTrackingScope(ctx)` helper, mirroring the existing `resolveUrlInspectorScope` pattern - `src/controllers/elements.js`
- suggestion: Add a one-line comment on the empty-projectIds early return explaining why it short-circuits (prevents querying workspace-wide unscoped data, which would leak other brands) - `src/controllers/elements.js:1536`
</details>

Note: ReBAC classification is correct - `:spaceCatId` and `:brandId` reuse existing aliases, `llmo/can_view` capability matches sibling routes and is in the published catalog. Security review found no vulnerabilities - auth, input validation, and trust boundaries follow established patterns.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 3m 51s | <b>Cost:</b> $6.11 | <b>Commit:</b> <code>9b6f75f4d44cef47460c15abd4b5e3ac828dcffe</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: src/support/elements/definitions/competitor-summary.js (hunk 1 above)
Comment (aniham) [reply src/support/elements/definitions/competitor-summary.js]: Done — added `test/support/elements/definitions/competitor-summary.test.js` covering the happy-path sum-per-competitor case (including the y__mentions field-reuse behavior you flagged), brand exclusion (case-insensitive), sort order, empty/missing input, and non-numeric value coercion. 100% line/branch coverage on the transform; removed the `c8 ignore` accordingly. Controller/service-layer coverage remains deferred, matching the sibling POC endpoints in this file.

**Comment by aniham** [comment]
Addressed the review's must-fix items (pushed in d9f9ada1):

1. **Unit tests for `transformCompetitorSummary`** — added, 100% line/branch coverage, `c8 ignore` removed for that function (replied inline).
2. **Missing OpenAPI spec** — added the `v2-serenity-competitor-summary` path item in `serenity-api.yaml`, a `SerenityCompetitorSummary` schema, and the `api.yaml` reference, modeled directly on the sibling `market-tracking-trends` endpoint. Also added its fixture to `test/openapi-contract/serenity-api.test.js` (contract-tested against the schema, passing).
3. **Missing integration tests** — investigated and pushing back on doing this here: none of the sibling Elements-backed brand-presence endpoints (`getStats`, `getMarketTrackingTrends`, `listSentimentOverview`, `getUrlInspectorStats`, etc.) have IT coverage in `test/it/` — that whole family goes through the Elements API transport, not the Semrush-mock-backed harness `test/it/shared/tests/serenity.js` already covers (prompts/markets/tags/workspace lifecycle). Closing this needs new IT mock infrastructure for the family as a whole, which is a bigger, separate effort rather than something to bolt onto one new endpoint. Happy to file a follow-up ticket for that if useful.

Also picked up the non-blocking suggestion to comment the empty-`projectIds` early return. Left the `resolveMarketTrackingScope`-style extraction suggestion alone for now since it'd mean refactoring two pre-existing, already-shipped handlers (`getMarketTrackingTrends`, `getStats`) that this PR didn't otherwise touch — seems better as its own focused refactor PR than mixed into this one.

**Review by vivesing — APPROVED** [review]
